### PR TITLE
Modify Scale() and ScaleN() to round and drop all-zero samples when scaling

### PR DESCRIPTION
**Pull request by nolanmar511**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,6 +6,7 @@ require (
 	github.com/chzyer/logex v1.1.10 // indirect
 	github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e
 	github.com/chzyer/test v0.0.0-20180213035817-a1ea475d72b1 // indirect
+	github.com/google/go-cmp v0.5.2
```
Comment (aalexand) [inline go.mod:9]: I wonder if it's OK to add new dependencies in terms of pprof being a part of golang distribution.

**Review by aalexand — DISMISSED** [review]

**Review by aalexand — COMMENTED** [review]
Looks like CI failed.

File: go.mod (hunk 1 above)
Comment (nolanmar511) [reply go.mod]: Done.

**Review by aalexand — DISMISSED** [review]

File: profile/profile.go [hunk 2]
```diff
@@ -712,7 +713,8 @@ func (s *Sample) DiffBaseSample() bool {
 	return s.HasLabel("pprof::base", "true")
 }
 
-// Scale multiplies all sample values in a profile by a constant.
+// Scale multiplies all sample values in a profile by a constant and keeps
```
Comment (kalyanac) [inline profile/profile.go:716]: and only keeps samples ...

**Comment by kalyanac** [comment]
Restarted the failed test.

File: profile/profile.go (hunk 2 above)
Comment (nolanmar511) [reply profile/profile.go:716]: done.

**Comment by nolanmar511** [comment]
I think the two test failures are unrelated to this change; I don' t think re-running will help.

See https://github.com/google/pprof/pull/560#issuecomment-699088881.

**Comment by aalexand** [comment]
@nolanmar511 Interestingly there are still CI failures.

**Comment by nolanmar511** [comment]
Just looking at this. The failures are also happening on the main branch.
I've filed https://github.com/google/pprof/issues/585.

**Comment by nolanmar511** [comment]
Tests are all passing now. PTAL

**Review by aalexand — APPROVED** [review]

**Pull request merged**